src/datastore/pg-write-store.ts

```diff
@@ -1963,13 +1965,33 @@ export class PgWriteStore extends PgStore {
     for (const txId of txIds) {
       logger.verbose(`Restoring mempool tx: ${txId}`);
     }
-    const updateResults = await sql<{ tx_id: string }[]>`
+
+    const updatedRows = await sql<{ tx_id: string }[]>`
       UPDATE mempool_txs
       SET pruned = false
       WHERE tx_id IN ${sql(txIds)}
       RETURNING tx_id
     `;
-    const restoredTxs = updateResults.map(r => r.tx_id);
+
+    const restoredTxs = updatedRows.map(r => r.tx_id);
+
+    // txs that didnt exist in the mempool need to be inserted into the mempool
+    if (updatedRows.length < txIds.length) {
+      const updatedTxs = updatedRows.map(r => r.tx_id);
+      const txsRequiringInsertion = txIds.filter(txId => !updatedTxs.includes(txId));
+
+      const txs: TxQueryResult[] = await sql`
+        SELECT * FROM txs
```

Comment:
These look equivalent to me? Why the change?

Reply:
Generally the usage of `SELECT *` is discouraged because it can cause prevent various optimizations in the sql engine and can result in unnecessary inefficiencies. We'd of course need to run benchmarks for this specific query on a production data set to validate that. 

There's also the simple case of: later on if we add additional columns to this table that are not needed for many queries, then those with a `SELECT *` will have a perf hit due to additional unnecessary db read and network transmission overhead.

Also, in this repo, we generally err on the side of more explicit when it comes to nits like this.